cincinnati/src/plugins/internal/metadata_fetch_quay.rs

```diff
@@ -0,0 +1,195 @@
+            |(release_id, release_version, manifestref): (ReleaseId, String, String)| -> Fallible<()> {
+                let mut rt = self::tokio::runtime::current_thread::Runtime::new()
+                    .context("could not create a Runtime")?;
+                let quay_labels: Vec<(String, String)> = rt
+                    .block_on(
+                        self.client
+                            .get_labels(
+                                self.repo.clone(),
+                                manifestref.clone(),
+                                Some(self.label_filter.clone()),
+                            )
+                            .map(|labels| labels.into_iter().map(Into::into).collect()),
+                    )
+                    .context(format!(
+                        "[{}] could not fetch quay labels",
+                        &release_version
+                    ))?;
+
+                info!(
+                    "[{}] received {} label(s)",
+                    &release_version,
+                    &quay_labels.len()
+                );
+
+                let metadata = graph.get_metadata_as_ref_mut(&release_id)?;
+
+                for (key, value) in quay_labels {
+                    let warn_msg = if metadata.contains_key(&key) {
+                        Some(format!(
+                            "[{}] key '{}' already exists. overwriting with value '{}'. ",
+                            &release_version, &key, &value
+                        ))
+                    } else {
+                        None
+                    };
+
+                    trace!(
+                        "[{}] inserting ('{}', '{}')",
+                        &release_version,
+                        &key,
+                        &value
+                    );
+
+                    if let Some(previous_value) = metadata.insert(key, value) {
+                        warn!(
+                            "{}previous value: '{}'",
+                            warn_msg.unwrap_or_default(),
+                            previous_value
+                        );
+                    };
+                }
+
+                Ok(())
+            },
+        )?;
+
+        Ok(InternalIO { graph, parameters })
+    }
+}
+
+impl QuayMetadataFetchPlugin {
+    pub fn try_new(
+        repo: String,
+        label_filter: String,
+        manifestref_key: String,
+        api_token_path: Option<&PathBuf>,
+        api_base: String,
+    ) -> Fallible<Self> {
+        let api_token =
+            quay::read_credentials(api_token_path).expect("could not read quay API credentials");
+
+        let client: quay::v1::Client = quay::v1::Client::builder()
+            .access_token(api_token.map(|s| s.to_string()))
+            .api_base(Some(api_base.to_string()))
+            .build()?;
+
+        Ok(Self {
+            client,
+            repo,
+            label_filter,
+            manifestref_key,
+        })
+    }
+}
+
+mod async {
```

Comment:
would uppercase `Async` work?

Reply:
Since I don't want to blow the scope of this PR let's get all of it in without async support. Therefore I've removed the mod alltogether.

Reply:
Answering first question, I'm usually settling for `asynchro`.

We discussed out-of-band that we will likely need some invasive work for the "async plugins" topic, which is mostly relevant for external plugins. It will be tackled in a followup PR as this one is already big enough.